### ISSUE TITLE
eox: fast EO to XMIR(XML) translator

### ISSUE THREAD

**Issue by yegor256** [objectionary/ideas#31]
Now we have a [translator](https://github.com/objectionary/eo) from EO to XMIR (XML), which is written in Java as an ANTLR4 parser with a collection of XSL stylesheets. It works, but very slowly. It takes a minute or so to translate a single `.eo` file to `.xmir` file. We can make a much faster alternative, written in Rust. It should use the existing [`Program.g4`](https://github.com/objectionary/eo/blob/0.28.11/eo-parser/src/main/antlr4/org/eolang/parser/Program.g4) grammar (not re-writing it!) and [existing XSL stylesheets](https://github.com/objectionary/eo/tree/0.28.11/eo-parser/src/main/resources/org/eolang/parser) (again, not making copies of them, but using them!). It should work as a command line tool (installable by [Cargo](https://crates.io)):

```bash
$ cargo install eox
$ eox -o hello.xmir hello.eo
```

It should be possible to configure the behavior of it, for example by removing certain stylesheets from the train:

```
$ eox -o hello.xmir --skip:add-refs hello.eo
```

Also, should be possible to append additional stylesheets to the train:

```
$ eox -o sodg.xml --append:sodg --append:sodg-to-text hello.eo
```

**Comment by l3r8yJ** [comment]
@yegor256 is this issue relevant?

**Comment by yegor256** [comment]
@l3r8yJ I would keep this one on hold, since we don't know yet what would be the language to implement the next version of EO compiler

**Comment by l3r8yJ** [comment]
@yegor256 have you seen [zig](https://ziglang.org/)? It's like a "more convenient" C — VM/compilers/translators can be easily written in this language

**Comment by yegor256** [comment]
@l3r8yJ yes, we've thought about Zig. Also about Haskell. However, now it's too early to make this decision.
